MIGRATION.md

```diff
@@ -1,5 +1,117 @@
 # Migration guide
 
+## Migrating to JSON Forms 3.6
+
+### Combinator (anyOf & oneOf) index selection now uses a heuristic instead of AJV
+
+In this update, we have eliminated the direct usage of AJV to determine the selected subschema for combinator renderers.
+To achieve this, the algorithm in `getCombinatorIndexOfFittingSchema` and with this `mapStateToCombinatorRendererProps` was changed.
+Thus, custom renderers using either method might have behavior changes.
+This rework is part of an ongoing effort to remove mandatory usage of AJV from JSON Forms.
+
+Before this change, AJV was used to validate the current data against all schemas of the combinator.
+This was replaced by using a heuristic which tries to match the schema via an identification property
+against a `const` entry in the schema.
+
+The identification property is determined as follows in descending order of priority:
+
+1. The schema contains a new custom property `x-jsf-type-property` next to the combinator to define the identification property.
+2. The data has any of these properties: `type`, `kind`, `id`. They are considered in the listed order.
+3. The data has any string or number property. The first encountered one is used.
```

Comment:
We should remove this as this is kind of arbitrary. Potentially we could search for a `const` property with limitations on nested level etc. Could also be a followup as this is not mission critical.

Reply:
I agree :+1: 
As discussed, we should also remove the `id` property from the second point because it seems likely that this is used for other purposes.